kedro-datasets/kedro_datasets_experimental/opik/opik_prompt_dataset.py

```diff
@@ -0,0 +1,596 @@
+        return data
+    return "\n".join(msg["content"] for msg in data)
+
+
+class OpikPromptDataset(AbstractDataset):
+    """Kedro dataset for managing prompts with Opik versioning and synchronisation.
+
+    This dataset provides seamless integration between local prompt files (JSON/YAML)
+    and Opik's prompt management system, supporting version control, metadata tracking,
+    and different synchronisation policies.
+
+    On save: Creates a new version of prompt in Opik with the local data.
+    On load: Synchronises based on sync_policy and returns raw Opik object
+    (sdk mode) or LangChain ChatPromptTemplate (langchain mode).
+
+    Sync policies:
+    - local: Local file takes precedence (default). Load_args are ignored with warning
+      since local files are the source of truth.
+    - remote: Opik version takes precedence. Load_args are respected if supported.
+    - strict: Error if local and remote differ. Load_args are respected if supported.
+
+    Examples:
+        Using catalog YAML configuration:
+
+        ```yaml
+        # Local sync policy - local files are source of truth
+        customer_prompt:
+          type: kedro_datasets_experimental.opik.OpikPromptDataset
+          filepath: data/prompts/customer.json
+          prompt_name: customer_support_v1
+          prompt_type: chat
+          credentials: opik_credentials
+          sync_policy: local
+          mode: langchain
+
+        # Remote sync policy - Opik versions are source of truth
+        production_prompt:
+          type: kedro_datasets_experimental.opik.OpikPromptDataset
+          filepath: data/prompts/production.yaml
+          prompt_name: customer_support_v1
+          sync_policy: remote
+          mode: sdk
+        ```
+
+        Using Python API:
+
+        ```python
+        from kedro_datasets_experimental.opik import OpikPromptDataset
+
+        # Create dataset for chat prompt
+        dataset = OpikPromptDataset(
+            filepath="data/prompts/customer_support.json",
+            prompt_name="customer_support_v1",
+            prompt_type="chat",
+            credentials={"api_key": "opik_...", "workspace": "my-workspace"}  # pragma: allowlist secret
+        )
+
+        # Load prompt as LangChain ChatPromptTemplate
+        prompt_template = dataset.load()
+        formatted = prompt_template.format(question="How are you?")
+
+        # Save with metadata
+        messages = [
+            {"role": "system", "content": "You are a helpful assistant."},
+            {"role": "user", "content": "Hello, {question}"}
+        ]
+        dataset.save(messages)
+        ```
+    """
+
+    def __init__(  # noqa: PLR0913
+        self,
+        filepath: str,
+        prompt_name: str,
+        prompt_type: Literal["chat", "text"] = "text",
+        sync_policy: Literal["local", "remote", "strict"] = "local",
+        mode: Literal["langchain", "sdk"] = "sdk",
+        credentials: dict[str, Any] | None = None,
```

Comment:
Are credentials optional? I would expect this dataset doesn't work without them just like the `LangfusePromptDataset`